contrib/sarplus/README.md

```diff
@@ -0,0 +1,149 @@
+# SARplus (preview)
+pronounced sUrplus as it's simply better if not best!
+
+[![Build Status](https://dev.azure.com/best-practices/recommenders/_apis/build/status/contrib%20sarplus?branchName=master)](https://dev.azure.com/best-practices/recommenders/_build/latest?definitionId=107&branchName=master)
+[![PyPI version](https://badge.fury.io/py/pysarplus.svg)](https://badge.fury.io/py/pysarplus)
+

```

Comment:
what is our rule going forward to have an algo in contrib folder vs having it under models? Maybe we should have a readme in contrib to explain this

Reply:
the idea is that contrib acts as a way to house independent algorithms where there is no current need to integrate them into the reco_utils package. we may want to explore leveraging this approach further for more algorithms to provide separation of dependencies. the downside is the testing and use of utils becomes challenging. the upside is this might be an easy way to quickly bring in algorithms / tools and mature them until they can be fully integrated.

but yes, agree on adding a readme explaining this

Reply:
I think it would be good to discuss the conditions in the next reco meeting. Some ideas discussed:
- When we want a separation of dependencies
- When  there is no current need to integrate them into the reco_utils package
- When the package has non-python compilation requirements

maybe @anargyri has more ideas here

Reply:
I have already mentioned some alternative approaches followed by other repos. E.g. SciPy and Scikit-Learn allow for add-ons / toolkits or compatible repositories:
https://www.scipy.org/scikits.html
https://github.com/scikit-learn-contrib
Surprise is a scikit, for example. 

We may want to consider these alternatives, as they offer better separation. 
Also, has anyone seen before a pypi package that is a subfolder of a repo (which could also be on pypi at some point in the future)? This setup seems unusual to me.
Another issue is that now some information is missing from the centralized point (front page) and so sarplus can be easily missed by someone who is not aware of the `contrib` structure. So, if sarplus is part of the repo, then it should be presented in the same way as the algorithms in `reco_utils`.

Reply:
> sarplus is closely related to SAR and this repo, so it makes sense to have both of them together.

the reason to not put them together was because it used non-python code?